Data/Aeson/Types/Internal.hs

```diff
@@ -549,6 +549,10 @@ data SumEncoding =
     -- by the encoded value of that field! If the constructor is not a
     -- record the encoded constructor contents will be stored under
     -- the 'contentsFieldName' field.
+  | UntaggedValue
+    -- ^ Constructor won't be encoded. The contents of the constructor will be
+    -- encoded as if the type had single constructor.  JSON encodings have to be
+    -- disjoint for decoding to work properly.
```

Comment:
I think this can be clarified a bit, how about "Constructor **names** won't be encoded. **Instead only** the contents of the constructor will be encoded, as if the type had a single constructor"?